Improve List demo description - list editing (#542)
DevExpress/devextreme-demos: JSDemos/Demos/List/ListEditingAndAPI/description.md

> The [List](/Documentation/ApiReference/UI_Components/dxList/) displays a collection of items as a scrollable list. To create a list, pass an array of items to the [dataSource](/Documentation/ApiReference/UI_Components/dxList/Configuration/#dataSource) property. 

Reviewer: I don't think you need this part here. The demo is not an introduction to lists, it's shows very specific functionality. We don't need to start every demo with a widget definition and basic properties.

Author: It may be better to keep this a text as this demo is the first in the series. This text will serve as an introduction to the List demo series as well as a reminder to users as to how to create a list.

> To allow users to delete items, set the [allowItemDeleting](/Documentation/ApiReference/UI_Components/dxList/Configuration/#allowItemDeleting) property to **true**. Use the "Allow deletion" checkbox under the List to see how this property works.

Reviewer 2: "to see how this property works" instead of saying what you should see, it is better to state what something does.

Author: Could you provide an example, please?

Reviewer 2: The "Allow deletion" checkbox under the List does this or that... (what does it do? Saying that it lets you see how something works does not explain what this checkbox does) Does it change something?

> This demo illustrates item deletion. To allow users delete items, set the [allowItemDeleting](/Documentation/ApiReference/UI_Components/dxList/Configuration/#allowItemDeleting) property to **true**. Click the Allow deleting checkbox under the List to see how this property works.

Reviewer: If you change demo title as I suggest, then you don't really need the first sentence. 

You need to add some formatting to the checkbox option - enclose in quotes or use bold formatting. Otherwise I see that "Allow" is capitalized, but I don't know if "deleting" is still part of the option name. 

Allow deleting is not a great name anyway - see our guidelines on INGs. Allow deletion would be better. 

```suggestion
To allow users to delete items, set the [allowItemDeleting](/Documentation/ApiReference/UI_Components/dxList/Configuration/#allowItemDeleting) property to **true**. Use the "Allow deletion" checkbox under the List to see how this property works.
```

> Use the "Deletion UI type" drop-down menu under the List widget to try different deletion behaviors described below. Menu items correspond to [itemDeleteMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemDeleteMode) property values. 

Reviewer 2: For example,

```suggestion
The "Deletion UI type" drop-down menu under the List widget allows you to switch between the different deletion behaviors described below. Menu items correspond to [itemDeleteMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemDeleteMode) property values. 
```

> The **List** supports several deletion modes listed below. Use the [itemDeleteMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemDeleteMode) property to set one of the modes. In this demo, you can select a deletion mode from the itemDeleteMode drop-down menu under the List.

Reviewer: Too many times you need to refer to "modes". Part of the problem is that you start with a format introduction "the component supports". These things don't require much explanation. If you tell people they can select modes, it's understood that the widget supports different modes. So no need in sentence #1. 

Again, I would change "deleting" in the control name to get rid of ING. 

```suggestion
Use the "Deletion UI type" drop-down menu under the List widget to try different deletion behaviors described below. Menu items correspond to [itemDeleteMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemDeleteMode) property values. 
```

> - *"static"*    
> Click an **X** button next to a list item to delete it. This is the default mode for desktops. 

Reviewer 3: ```suggestion
- *"static"* (default for desktops)    
Click an X button next to a list item to delete it. 
```

- *"toggle"*    

> Click a **minus** icon next to a list item to display the Delete button. 

Reviewer: Just to let you know, when I select this mode, I don't see "minus" icons. I know where they should be and I can click that area, but it's empty.

Author: We are aware of this issue and developers are handling it. By the way, the problem goes away on its own when the select functionality is removed from this demo.

- *"slideButton"*    

> Swipe left or right to show the Delete button next to the list item. The content stays in place.  

Reviewer 2: "The content stays in place" Do you mean the content does not move? As in, the content is not affected?

Author: Yes. In some cases, list items are moved off-canvas but in this case, they don't move

Reviewer 2: ```suggestion
Swipe left or right to show the Delete button next to the list item. The content of the list item does not move.  
```

> Slide list items left to show the **delete** button. In this case, the list item is moved off the canvas. 

Reviewer 3: ```suggestion
Slide list items left to show the Delete button. In this case, the list item is moved off the canvas. 
```

> Swipe left or right to show the Delete button next to the list item. The content stays in its place.  

Reviewer: ```suggestion
Swipe left or right to show the Delete button next to the list item. The content stays in place.  
```

- *"slideItem"* (default for iOS)    

> Slide list items left. Item content shifts and the Delete button appears. 

Reviewer 2: ```suggestion
Slide list items left. The content of the list item shifts to make space for the Delete button. 
```

> Slide list items left to show the Delete button. In this case, the list item is moved off the canvas. 

Reviewer: ```suggestion
Slide list items left. Item content shifts and the Delete button appears. 
```

- *"toggle"*    

> Click a **minus** icon next to a list item to display the **delete** button. 

Reviewer 3: ```suggestion
Click a **minus** icon next to a list item to display the Delete button. 
```

- *"context"*    

> Right-click or touch and hold the list item to call a context menu that contains the **delete** option.    

Reviewer 3: ```suggestion
Right-click or touch and hold the list item to call a context menu that contains the Delete option.    
```

> The [List](/Documentation/ApiReference/UI_Widgets/dxList/) displays a collection of items as a scrollable list. To create a list, pass an array of items to **dataSource**. 

Reviewer 3: ```suggestion
The [List](/Documentation/ApiReference/UI_Widgets/dxList/) displays a collection of items as a scrollable list. To create a list, pass an array of items to the [dataSource](/Documentation/ApiReference/UI_Components/dxList/Configuration/#dataSource) property. 
```

> This demo illustrates deletion. The UI consists of the **List** and the "Options" section. 
> 
> To allow users delete items, set the [allowItemDeleting](/Documentation/ApiReference/UI_Widgets/dxList/Configuration/#allowItemDeleting) property to *true*. Use the checkbox in the "Options" section to see how it works.

Reviewer 3: I suggest removing the sentence about the UI because it seems unnecessary if we don't talk about elements in the Options section.

```suggestion
This demo illustrates item deletion. To allow users delete items, set the [allowItemDeleting](/Documentation/ApiReference/UI_Widgets/dxList/Configuration/#allowItemDeleting) property to **true**. Click the Allow deleting checkbox under the List to see how this property works.
```

> The **List** supports three default and three additional deletion modes. You can see them in action in the "itemDeleteMode" dropdown menu of the "Options" section.

Reviewer 3: Let's give the name of the property as early in the description as possible.

```suggestion
The **List** supports several deletion modes listed below. Use the [itemDeleteMode](/Documentation/ApiReference/UI_Widgets/dxList/Configuration/#itemDeleteMode) property to set one of the modes. In this demo, you can select a deletion mode from the itemDeleteMode drop-down menu under the List.
```

> **Default modes**

Reviewer 3: The grouping is unnecessary. We can indicate that one or another mode is default for a platform like this:

`- *"slideItem"* (default for iOS)`

Let's get back to a plain list.

    
When you select *"static"* mode, click an **X** button next to a list item to delete it. This is the default mode for desktops. To see the default iOS deletion mode, select *"slideItem"*. Slide list items left to show the **delete** button. In this case, the list item is moved off the canvas. To try the deletion mode on Android, select *"swipe"* and swipe items to delete them.

**Additional modes**

- *"slideButton"*    

> Users can swipe right to show the **delete** button next to the list item. The content stays in its place.    

Reviewer 3: ```suggestion
Swipe left or right to show the Delete button next to the list item. The content stays in its place.    
```

> - *"Toggle"*    

Reviewer 3: ```suggestion
- *"toggle"*    
```

Click a **minus** icon next to a list item to display the **delete** button. 

- *"context"*    
Right-click or touch and hold the list item to call a context menu that contains the **delete** option.    

> Use the [itemDeleteMode](/Documentation/ApiReference/UI_Widgets/dxList/Configuration/#itemDeleteMode) property to set the deletion mode.
> 
> When users delete a list item, it is removed from the "Selected tasks" section. This is done by a function that handles the [onItemDeleted](/Documentation/ApiReference/UI_Widgets/dxList/Configuration/#onItemDeleted) event.

Reviewer 3: The `itemDeleteMode` is mentioned before the list now.
Selected tasks are going to be removed as we agreed.

```suggestion
```
